blocks/examples/mnist/train.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+
+from theano import tensor
+
+from blocks.datasets import DataStream
+from blocks.datasets.mnist import MNIST
+from blocks.datasets.schemes import SequentialScheme
+from blocks.algorithms import GradientDescent, DefaultStepRule
+from blocks.bricks import MLP, Tanh, Identity
+from blocks.bricks.cost import BinaryCrossEntropy
+from blocks.main_loop import MainLoop
+from blocks.extensions import FinishAfter, Printing
+
+
+def main():
+    mlp = MLP([Tanh(), Identity()], [784, 100, 10])
+    x = tensor.matrix('features')
+    y = tensor.lmatrix('targets')
+    cost = BinaryCrossEntropy().apply_for_indices(y - 1, mlp.apply(x))
```

Comment:
MNIST doesn't use `BinaryCrossEntropy` (which suggests that there are two labels), but `CategoricalCrossEntropy` (which needs to be written still as a wrapper of `theano.tensor.nnet.categorical_crossentropy` instead of being a method of `BinaryCrossEntropy`).